RoddyCore/src/de/dkfz/roddy/execution/jobs/Job.groovy

```diff
@@ -333,6 +338,100 @@ class Job extends BEJob<BEJob, JobResult> {
         path
     }
 
+
+    /**
+     * Stores a new job jobState info to an execution contexts job jobState log file.
+     *
+     * @param job
+     */
+    void appendToJobStateLogfile(BatchEuphoriaJobManager jobManager, ExecutionContext executionContext, BEJobResult res, OutputStream out = null) {
+        // Ugly, but works. Java/Groovy's missing runtime dispatch on method arguments sucks.
+        if (jobManager instanceof ClusterJobManager) {
+            appendToJobStateLogfileForCluster(executionContext, res)
+        } else if (jobManager instanceof DirectSynchronousExecutionJobManager) {
+            appendToJobStateLogfileLocal(executionContext, res, out)
+        } else {
+            throw new NotImplementedException()
+        }
+    }
+    private static String jobStateInfoLine(String jobId, String code, String millis) {
+        return String.format("%s:%s:%s", jobId, code, millis)
+    }
+
+
+    void appendToJobStateLogfileForCluster(ExecutionContext executionContext, BEJobResult res) {
+        if (res.wasExecuted) {
+            def job = res.command.getJob()
+            String jobInfoLine = getJobStateInfoLineForClusterJob(job)
+            if (jobInfoLine != null)
```

Comment:
I don't think so, if we don't want to add more lines to the jobStateLogfile.

In principle I do not have any problems with adding new lines to the jobStateLogfile (but it needs to be communicated to the OTP crowd). The jobStateLogFile would then even more become an event log (submitted, started, stopped-with-exit-code, etc.).